Makefile

```diff
@@ -48,18 +47,11 @@ ifeq ($(OPTIMIZE),yes)
 BASE_CXXFLAGS += -O2
 endif
 
-ifneq ($(USE_LUA), yes)
-USE_LUA=no
-endif
-
 BASE_CXXFLAGS += -Wall -Werror
 
 ifneq (,$(findstring clang, `$(CXX)`))
 SANITIZE_UNDEFINED=
 BASE_CXXFLAGS += -Qunused-arguments -Wno-unknown-warning-option -Wno-deprecated-register
-ifeq ($(USE_LUA), yes)
-BASE_CXXFLAGS += -Wno-pointer-bool-conversion -Wno-parentheses-equality
-endif
 else ifneq (, $(findstring g++, `$(CXX)`))
```

Comment:
Actually missed this carrying over to an else now. Will reinvestigate.

Reply:
This actually made no sense anyway as the `endif` is before the else. Also the second one does not terminate right now so I'm very confused as to what's what. Let's see.

Reply:
Actually this is fine. It's just confusing how there are nested `if` blocks. The else below here is for the `ifneq` above and removing this `ifeq` block is fine. Compiler specific setup for warning flags. IMO once we have the CI going where we should start getting all the compiler dependent flags out of here and make the Makefile as simple as possible.